src/services/dumpers/agent-nodejs.ts

```diff
@@ -0,0 +1,209 @@
+        dependencies.tedious = '^6.4.0';
+      }
+    }
+
+    const pkg = {
+      name: this.toValidPackageName(appName),
+      version: '0.0.1',
+      private: true,
+      main: 'index.js',
+      scripts: {
+        'start:watch': 'nodemon',
+        start: 'node ./index.js',
+      },
+      nodemonConfig: {
+        ignore: ['./forestadmin-schema.json'],
+      },
+      dependencies,
+      devDependencies: {
+        nodemon: '^2.0.12',
+      },
+    };
+
+    this.writeFile('package.json', `${JSON.stringify(pkg, null, 2)}\n`);
+  }
+
+  writeIndex(dbDialect: string) {
+    const isMongoose = dbDialect === 'mongodb';
+
+    const context = {
+      isMongoose,
+      isMySQL: dbDialect === 'mysql',
+      isMSSQL: dbDialect === 'mssql',
+      isMariaDB: dbDialect === 'mariadb',
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      datasourceImport: isMongoose
+        ? `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst connection = require('./mongoose-models');`
+        : `const { createSqlDataSource } = require('@forestadmin/datasource-sql');`,
+      datasourceCreation: isMongoose
+        ? 'createMongooseDataSource(connection, {})'
+        : `
+    createSqlDataSource({
+      uri: process.env.DATABASE_URL,
+      schema: process.env.DATABASE_SCHEMA || 'public',
+      ...dialectOptions,
+    }),
+  `,
+    };
+
+    this.copyHandleBarsTemplate('index.hbs', 'index.js', context);
+  }
+
+  private writeDotEnv(
+    dbConfig: DbConfig,
+    appPort: number,
+    forestEnvSecret: string,
+    forestAuthSecret: string,
+  ) {
+    const dbUrl = this.buildDatabaseUrl(dbConfig);
+    const context = {
+      dbUrl,
+      dbSsl: dbConfig.dbSsl || false,
+      dbSchema: dbConfig.dbSchema !== '' ? dbConfig.dbSchema : false,
+      appPort,
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      forestEnvSecret,
+      forestAuthSecret,
+      hasDockerDbUrl: false,
+      dockerDbUrl: '',
+    };
+
+    if (!this.isLinuxOs) {
+      context.hasDockerDbUrl = true;
+      context.dockerDbUrl = dbUrl.replace('localhost', 'host.docker.internal');
+    }
+
+    this.copyHandleBarsTemplate('env.hbs', '.env', context);
+  }
+
+  private writeGitignore() {
+    this.writeFile('.gitignore', 'node_modules\n.env\n');
+  }
+
+  private writeTypings() {
+    this.writeFile('typings.ts', '/* eslint-disable */\nexport type Schema = any;\n');
+  }
+
+  private writeDockerignore() {
+    this.writeFile('.dockerignore', 'node_modules\nnpm-debug.log\n.env\n');
+  }
+
+  private writeDockerfile() {
+    this.copyHandleBarsTemplate('Dockerfile.hbs', 'Dockerfile');
+  }
+
+  private writeDockerCompose(config: Config) {
+    const dbUrl = `\${${this.isLinuxOs ? 'DATABASE_URL' : 'DOCKER_DATABASE_URL'}}`;
+    const forestServerUrl = this.env.FOREST_URL_IS_DEFAULT ? false : `\${FOREST_SERVER_URL}`;
```

Comment:
After reading this multiple time, I don't really get with we inject something "complicated" in FOREST_SERVER_URL to always re-manipulate it each time.